thorlcr/graph/thgraphmaster.cpp

```diff
@@ -2724,22 +2727,24 @@ void CMasterGraph::handleSlaveDone(unsigned node, MemoryBuffer &mb)
     }
 }
 
-void CMasterGraph::getFinalProgress()
+void CMasterGraph::getFinalProgress(bool aborting)
 {
     CMessageBuffer msg;
     mptag_t replyTag = queryJobChannel().queryMPServer().createReplyTag();
     msg.setReplyTag(replyTag);
     msg.append((unsigned)GraphEnd);
     msg.append(job.queryKey());
     msg.append(queryGraphId());
-    jobM->broadcast(queryNodeComm(), msg, masterSlaveMpTag, LONGTIMEOUT, "graphEnd", NULL, true);
+    // If aborted, some slaves may have disconnnected/aborted so don't wait so long
+    unsigned timeOutPeriod = aborted ? SHORTTIMEOUT : LONGTIMEOUT;
```

Comment:
Double checking that this is correct to test aborted rather than aborting.

Reply:
You're right, it should be aborting.